.azure-pipelines/azure-pipelines-backend.yml

```diff
@@ -3,43 +3,95 @@ trigger:
     include:
       - master
   paths:
-    exclude: 
+    exclude:
       - /.azure-pipelines/azure-pipelines-frontend.yml
       - /packages/client/*
       - /packages/client-e2e/*
 
 pool:
   vmImage: 'ubuntu-latest'
 
-steps: 
+steps:
   - task: NodeTool@0
     inputs:
       versionSpec: '14.6.0'
     displayName: 'Install Node.js 14.6.0'
 
+  # This is included in package.json, but only under devDependencies, so this (production!) build won't see it.
+  # As a workaround, install it globally
+  - script: yarn global add wasm-pack
+    displayName: 'Install wasm-pack'
+
+  - task: Cache@2
+    inputs:
+      # Cache the tarred node_modules.xz file.
+      # Note that the cache key is the hash of of yarn.lock. If the lockfile changes, the cache gets invalidated.
+      key: 'v2 | yarn | "$(Agent.OS)" | $(Build.SourcesDirectory)/yarn.lock'
+      path: '$(Build.SourcesDirectory)/tarred_node_modules'
+      cacheHitVar: 'NodeCacheRestored'
+    displayName: 'Cache node_modules'
+
+  # Untar the cached node_modules folder if we had a cache hit
+  - script: tar -xf $(Build.SourcesDirectory)/tarred_node_modules/node_modules.tar.xz --directory=$(Build.SourcesDirectory) --verbose
+    condition: eq(variables['NodeCacheRestored'], 'true')
+    displayName: 'Untar node_modules.xz'
+
+  # Cache word_counter's output
   - task: Cache@2
     inputs:
-    # Note that the cache key is the hash of of yarn.lock. If the lockfile changes, the cache gets invalidated.
-      key: 'yarn | "$(Agent.OS)" | $(Build.SourcesDirectory)/yarn.lock'
-      path: '$(Build.SourcesDirectory)/node_modules'
-      cacheHitVar: 'CacheRestored'
-    displayName: "Cache node_modules"
+      key: 'cargo | "$(Agent.OS)" | $(Build.SourcesDirectory)/native/word_counter/Cargo.lock'
+      path: '$(Build.SourcesDirectory)/native/word_counter/target'
+      cacheHitVar: 'WordCounterCacheRestored'
+    displayName: 'Cache word_counter output'
 
-    # Install production-only packages if the cache doesn't exist, or is out of date.
+  # Cache html_sanitizer's output
+  - task: Cache@2
+    inputs:
+      key: 'cargo | "$(Agent.OS)" | $(Build.SourcesDirectory)/native/html_sanitizer/Cargo.lock'
+      path: '$(Build.SourcesDirectory)/native/html_sanitizer/target'
+      cacheHitVar: 'WordCounterCacheRestored'
+    displayName: 'Cache html_sanitizer output'
```

Comment:
This is gonna get repetitive as we add more native modules. We can probably split it out into a sub-task, and just pass in an array of module names. I'm pretty sure we can do that with Az DevOps.